### PR TITLE
[Feat] checkpoint_merger works on local models as well as ones that use safetensors

### PR DESCRIPTION
**Pull request by lstein**
The documentation for the community pipeline `checkpoint_merger.py` states that you can merge a model either using its repo_id or a path to the model directory on local disk. However, the latter does not seem to work because the model's directory gets passed to `snapshot_download()`.

This PR also adds the ability of this utility to merge models created with safetensors, and cleans up status reporting.

Apologies for the large number of commits (first time figuring out how to debug a community pipeline); just look at **Files changed**.

### CODE REVIEW

File: examples/community/checkpoint_merger.py [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 import torch
 
+import safetensors.torch
```
Comment (patrickvonplaten) [inline examples/community/checkpoint_merger.py:7]:
```suggestion
from diffusers.utils import is_safetensors_available

if is_safetensors_available():
    import safetensors.torch
```

File: examples/community/checkpoint_merger.py [hunk 2]
```diff
@@ -188,38 +193,56 @@ def merge(self, pretrained_model_name_or_path_list: List[Union[str, os.PathLike]
...
                     theta_0 = getattr(module, "state_dict")
                     theta_0 = theta_0()
 
                     update_theta_0 = getattr(module, "load_state_dict")
-                    theta_1 = torch.load(checkpoint_path_1, map_location="cpu")
-
-                    theta_2 = torch.load(checkpoint_path_2, map_location="cpu") if checkpoint_path_2 else None
+                    theta_1 = (
+                        safetensors.torch.load_file(checkpoint_path_1)
+                        if checkpoint_path_1.endswith(".safetensors")
```
Comment (patrickvonplaten) [inline examples/community/checkpoint_merger.py:226]:
```suggestion
                        if (is_safetensors_available() and checkpoint_path_1.endswith(".safetensors"))
```

File: examples/community/checkpoint_merger.py [hunk 3]
```diff
@@ -188,38 +193,56 @@ def merge(self, pretrained_model_name_or_path_list: List[Union[str, os.PathLike]
...
+                        else torch.load(checkpoint_path_1, map_location="cpu")
+                    )
+                    theta_2 = None
+                    if checkpoint_path_2:
+                        theta_2 = (
+                            safetensors.torch.load_file(checkpoint_path_2)
+                            if checkpoint_path_2.endswith(".safetensors")
```
Comment (patrickvonplaten) [inline examples/community/checkpoint_merger.py:233]:
```suggestion
                            if (is_safetensors_available() and checkpoint_path_2.endswith(".safetensors"))
```

**Comment by patrickvonplaten** [comment]
Thanks a lot for the PR @lstein !  This looks great to me, would just be nice to allow the script to also work when safetensors is not available.

**Comment by lstein** [comment]
Thank you for the review! I have committed all your suggestions.

**Comment by lstein** [comment]
I made a whitespace styling fix to the imports to satisfy isort. Not sure if policy requires re-review on your part.

**Review by patrickvonplaten — APPROVED** [review]
Perfect thanks a lot!

**Comment by Abhinay1997** [comment]
@lstein thank you for the fixes !

**Pull request merged**